pytorch_lightning/accelerators/tpu.py

```diff
@@ -46,4 +46,7 @@ def all_gather(self, tensor: torch.Tensor, group: Optional[Any] = None, sync_gra
         Return:
             A tensor of shape (world_size, batch, ...)
         """
-        return xm.all_gather(tensor, group=group, sync_grads=sync_grads)
+        # todo: Add support for backward with all_gather
+        if torch.distributed.is_initialized():
+            return xm.all_gather(tensor, group=group, sync_grads=sync_grads)
+        return tensor
```

Comment:
Code cov complaining about no test, probably want to add one to make sure this case is covered in single TPU!

Reply:
The test cover this part but coverage is currently not supported for TPU.